harness/determined/common/api/authentication.py

```diff
@@ -97,6 +97,7 @@ def _init_session(
 
         if (
             token is None
+            and requested_user is None
```

Comment:
What about the case where user `alice` creates a shell, then runs `det -u alice user whoami` in the shell?  Should `alice` need to provide a password?

It seems like this should be:
```python
if (
  token is None
  and requested_user in (None, util.get_det_username_from_env())
  and util.get_det_username_from_env() is not None
  and util.get_det_user_token_from_env() is not None
):
  ...

Reply:
Great catch, added!